core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.server
+
+import org.apache.kafka.common.protocol.ApiKeys
+import org.apache.kafka.common.requests.{ApiVersionsRequest, ApiVersionsResponse}
+
+abstract class AbstractApiVersionsRequestTest extends BaseRequestTest {
```

Comment:
Is this abstract base class worth it? We could either add this single method to `BaseRequestTest` or `TestUtils`.

Reply:
It's a bit more justifiable if we take the `validateApiVersionsResponse` method out of the `ApiVersionsRequestTest` companion.

Reply:
Maybe we should move this method to the companion? Nice to avoid long inheritance hierarchies.

Reply:
We could but it's a little more awkward since it requires an instance of `BaseRequestTest`. We could also add it to `BaseRequestTest`, but I was hoping to keep that class generic and to avoid leaking a much of api-specific noise into it.